pkg/kubectl/autoscale_test.go

```diff
@@ -0,0 +1,155 @@
+					Name: "foo",
+				},
+				Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
+					TargetCPUUtilizationPercentage: newInt32(80),
+					ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
+						Kind:       "kind",
+						Name:       "name",
+						APIVersion: "apiVersion",
+					},
+					MaxReplicas: int32(10),
+					MinReplicas: newInt32(1),
+				},
+			},
+			expectErr: false,
+		},
+		{
+			params: map[string]interface{}{
+				"scaleRef-kind":       "kind",
+				"scaleRef-name":       "name",
+				"scaleRef-apiVersion": "apiVersion",
+			},
+			expectErr: true,
+			reason:    "'name' is a required parameter",
+		},
+		{
+			params: map[string]interface{}{
+				"default-name":        "foo",
+				"scaleRef-kind":       "kind",
+				"scaleRef-name":       "name",
+				"scaleRef-apiVersion": "apiVersion",
+			},
+			expectErr: true,
+			reason:    "'max' is a required parameter",
+		},
+		{
+			params: map[string]interface{}{
+				"name":                "foo",
+				"min":                 "10",
+				"max":                 "1",
+				"scaleRef-kind":       "kind",
+				"scaleRef-name":       "name",
+				"scaleRef-apiVersion": "apiVersion",
+			},
+			expectErr: true,
+			reason:    "'max' must be greater than or equal to 'min'",
+		},
+		{
+			params: map[string]interface{}{
+				"name":                "foo",
+				"min":                 "1",
+				"max":                 "10",
+				"cpu-percent":         "",
+				"scaleRef-kind":       "kind",
+				"scaleRef-name":       "name",
+				"scaleRef-apiVersion": "apiVersion",
+			},
+			expectErr: true,
+			reason:    "cpu-percent must be an integer if specified",
+		},
+		{
+			params: map[string]interface{}{
+				"name":                "foo",
+				"min":                 "foo",
+				"max":                 "10",
+				"cpu-percent":         "60",
+				"scaleRef-kind":       "kind",
+				"scaleRef-name":       "name",
+				"scaleRef-apiVersion": "apiVersion",
+			},
+			expectErr: true,
+			reason:    "'min' must be an integer if specified",
+		},
+		{
+			params: map[string]interface{}{
+				"name":                "foo",
+				"min":                 "1",
+				"max":                 "bar",
+				"cpu-percent":         "90",
+				"scaleRef-kind":       "kind",
+				"scaleRef-name":       "name",
+				"scaleRef-apiVersion": "apiVersion",
+			},
+			expectErr: true,
+			reason:    "'max' must be an integer if specified",
+		},
+	}
+	generator := HorizontalPodAutoscalerV1{}
+	for i, test := range tests {
+		obj, err := generator.Generate(test.params)
+		if test.expectErr && err != nil {
+			continue
+		}
+		if !test.expectErr && err != nil {
+			t.Errorf("[%d] unexpected error: %v", i, err)
+		}
+		if test.expectErr && err == nil {
+			t.Errorf("Expect error, reason: %v, got nil", test.reason)
+		}
+		if !reflect.DeepEqual(obj.(*autoscalingv1.HorizontalPodAutoscaler), test.expected) {
+			t.Errorf("\n[%d] want:\n%#v\n[%d] got:\n%#v", i, test.expected, i, obj.(*autoscalingv1.HorizontalPodAutoscaler))
```

Comment:
It will be more helpful if you include test case name.
Test case name is better than index.